Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -343,8 +500,20 @@ void TextureMapperLayer::paintSelfAndChildren(TextureMapperPaintOptions& options
         }
     }
 
-    for (auto* child : m_children)
-        child->paintRecursive(options);
+    for (auto* child : m_children) {
+        // During flattening, already flattened children should not be painted
+        // because they are roots of their own render pass
```

Comment:
What is a render pass? I don't really understand the comment.

Reply:
It's a stage in overall rendering were part of the scene is render into intermediate buffer. Once all passes are generated then final scene can be produced. Here we skip painting of children which are already part of flattened buffers. Comment updated